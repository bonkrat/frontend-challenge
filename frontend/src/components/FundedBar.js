import React from "react";
import PropTypes from "prop-types";
import * as styles from "./FundedBar.styles";
import LinearProgress from "@material/react-linear-progress";

/**
 * Renders a bar showing how much the project has been funded.
 */
const FundedBar = ({ className: rootStyles, goal, funds }) => (
  <div className={rootStyles}>
    <span className={styles.fundedText}>
      {Math.floor((funds / goal) * 100)}% funded
    </span>

    <LinearProgress
      className={styles.bar}
      buffer={1}
      bufferingDots={false}
      progress={funds / goal}
    />
  </div>
);

FundedBar.propTypes = {
  /**
   * Classname to apply to the root.
   */
  className: PropTypes.string,
  /**
   * The amount needed to reach the goal of the campaign.
   */
  goal: PropTypes.number,
  /**
   * How much is currently funded.
   */
  funds: PropTypes.number
};

export default FundedBar;
