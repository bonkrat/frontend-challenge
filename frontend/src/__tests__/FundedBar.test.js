import React from "react";
import { shallow } from "enzyme";
import FundedBar from "../components/FundedBar";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  const props = {
    className: "foo",
    goal: 100,
    funds: 25
  };
  expect(toJson(shallow(<FundedBar {...props} />))).toMatchSnapshot();
});
