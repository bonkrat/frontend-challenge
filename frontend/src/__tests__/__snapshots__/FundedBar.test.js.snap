// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="foo"
>
  <span
    className="text-purple-900"
  >
    25
    % funded
  </span>
  <t
    buffer={1}
    bufferingDots={false}
    className="border-r rounded"
    closed={false}
    indeterminate={false}
    progress={0.25}
    reversed={false}
    tag="div"
  />
</div>
`;
